pymeasure/instruments/siglenttechnologies/siglent_spdbase.py

```diff
@@ -0,0 +1,278 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from pymeasure.instruments.instrument import Instrument
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_range
+                                              )
+from enum import IntFlag
+
+log = logging.getLogger(__name__)  # https://docs.python.org/3/howto/logging.html#library-config
+log.addHandler(logging.NullHandler())
+
+
+class SystemStatusCode(IntFlag):
+    """System status enums based on ``IntFlag``
+
+    Used in conjunction with :attr:`~.system_status_code`.
+
+        ======  ======
+        Value   Enum
+        ======  ======
+        256     WAVEFORM_DISPLAY
+        64      TIMER_ENABLED
+        32      FOUR_WIRE
+        16      OUTPUT_ENABLED
+        1       CONSTANT_CURRENT
+        0       CONSTANT_VOLTAGE
+        ======  ======
+
+    """
+
+    WAVEFORM_DISPLAY = 256  # bit 8 -- waveform display enabled
+    TIMER_ENABLED = 64  # bit 6 -- timer enabled
+    FOUR_WIRE = 32  # bit 5 -- four-wire mode enabled
+    OUTPUT_ENABLED = 16  # bit 4 -- output enabled
+    CONSTANT_CURRENT = 1   # bit 0 -- constant current mode
+    CONSTANT_VOLTAGE = 0   # bit 0 -- constant voltage mode
+
+
+class SPDBase(Instrument):
+    """ The base class for Siglent SPDxxxxX instruments.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            usb=dict(write_termination='\n',
+                     read_termination='\n'),
+            tcpip=dict(write_termination='\n',
+                       read_termination='\n'),
+            **kwargs
+        )
+
+    error = Instrument.measurement(
+        "SYST:ERR?",
+        """Read the error code and information of the instrument.
+
+        :type: str
```

Comment:
I've rarely seen this type annotation. For a getter (measurement), most of the time, the type does not matter, as you get the variable in the correct type. It's only important for the setter and there not often. Most of the time it is obvious (boolean, string, float, or int).
Additionally, you can write it in the text, if the type might be ambibous, e.g.:
```suggestion
        """Read the error code and information of the instrument as a string.
```

Reply:
In the case of this instrument, the error code (returned as an integer in a string format) is followed by a plain-text description of the error which has occurred. In the default case, the string "0 No Error" is the result which I agree is an unusual response. You're right that your suggestion is probably more readable as a sentence but this version of formatting should show up very clearly in the docs. The docs page for temptronic_base was my example for this